crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs

```diff
@@ -33,88 +35,93 @@ impl PeepholeReplaceKnownMethods {
-                    call_expr.span,
-                    call_expr,
-                    member,
-                    string_lit,
-                    ctx,
-                ),
-                // TODO: Implement the rest of the string methods
-                "substr" => None,
-                "substring" | "slice" => None,
-                "charAt" => {
-                    Self::try_fold_string_char_at(call_expr.span, call_expr, string_lit, ctx)
-                }
-                "charCodeAt" => None,
-                "replace" => None,
-                "replaceAll" => None,
-                _ => None,
-            };
-
-            if let Some(replacement) = replacement {
-                self.changed = true;
-                *node = replacement;
+        let Some(mem_expr) = call_expr.callee.as_member_expression() else { return };
+        let Some(string_lit) = Self::get_string_literal(mem_expr.object(), ctx) else { return };
+        let Some(method_name) = mem_expr.static_property_name() else { return };
+
+        let arguments = &call_expr.arguments;
+
+        #[expect(clippy::match_same_arms)]
+        let replacement = match method_name {
+            "toLowerCase" | "toUpperCase" | "trim" => {
+                let transformed_value = match method_name {
+                    "toLowerCase" => {
+                        Some(ctx.ast.string_literal(call_expr.span, string_lit.cow_to_lowercase()))
+                    }
+                    "toUpperCase" => {
+                        Some(ctx.ast.string_literal(call_expr.span, string_lit.cow_to_uppercase()))
+                    }
+                    "trim" => Some(ctx.ast.string_literal(call_expr.span, string_lit.trim())),
+                    _ => None,
+                };
+
+                transformed_value.map(|transformed_value| {
+                    ctx.ast.expression_from_string_literal(transformed_value)
+                })
             }
+            "indexOf" | "lastIndexOf" => Self::try_fold_string_index_of(
+                call_expr.span,
+                &string_lit,
+                method_name,
+                arguments,
+                ctx,
+            ),
+            // TODO: Implement the rest of the string methods
+            "substr" => None,
+            "substring" | "slice" => None,
+            "charAt" => Self::try_fold_string_char_at(call_expr.span, &string_lit, arguments, ctx),
+            "charCodeAt" => None,
+            "replace" => None,
+            "replaceAll" => None,
+            _ => None,
+        };
+
+        if let Some(replacement) = replacement {
+            self.changed = true;
+            *node = replacement;
         }
     }
 
     fn try_fold_string_index_of<'a>(
         span: Span,
-        call_expr: &CallExpression<'a>,
-        member: &StaticMemberExpression<'a>,
-        string_lit: &StringLiteral<'a>,
+        string_lit: &str,
+        method_name: &str,
+        arguments: &oxc_allocator::Vec<Argument<'a>>,
```

Comment:
I recommend not to use `Vec<Argument<'a>>` here.

There involves numerous arguments, which are same or different. For example, `indexOf` involves 0 to 2 arguments, while `join` involves 0 to 1. Using `Option` type and use `.get` or `.get_mut` method is proper.

The extracting process of the arguments should be involved in the parent function, which is the caller. Otherwise it is annoying to repeat these arguments.

Reply:
I simply passed the parameter list of `call_expr` to our string methods, and for those that do not require parameters, we can skip passing them.

As for where we should handle parameters, further discussion is needed. I think handling them outside of the method will become more complex.

I will restore it.